### PR TITLE
Updated handleAuthClick to return a promise

### PR DESCRIPTION
**Pull request by chantelc-mb**
Updated handleAuthClick to return a promise. Hoping to close this issue: handleAuthClick should return promise #67.

### CODE REVIEW

**Comment by nickakey** [comment]
Made this PR into your branch to update the readme for these changes :)

https://github.com/chantelc-mb/react-google-calendar-api/pull/1

File: src/ApiCalendar.ts [hunk 1]
```diff
@@ -79,12 +79,13 @@ class ApiCalendar {
   /**
    * Sign in Google user account
    */
-  public handleAuthClick(): void {
-    if (this.gapi) {
-      this.gapi.auth2.getAuthInstance().signIn();
-    } else {
-      console.log('Error: this.gapi not loaded');
-    }
+   public handleAuthClick(): any {
```
Comment (Kubessandra) [inline src/ApiCalendar.ts:82]:
```suggestion
   public handleAuthClick(): Promise<any> {
```

File: README.md [hunk 2]
```diff
@@ -46,8 +46,9 @@ https://console.developers.google.com/flows/enableapi?apiid=calendar.
 ```javascript
     /**
      * Sign in with a Google account.
+     * @returns {any} A Promise that is fulfilled with the GoogleUser instance when the user successfully authenticates and grants the requested scopes, or rejected with an object containing an error property if an error happened
      */
-    public handleAuthClick(): void
+    public handleAuthClick(): any
```
Comment (Kubessandra) [inline README.md:51]:
```suggestion
    public handleAuthClick(): Promise<any>
```

File: README.md [hunk 3]
```diff
@@ -74,6 +75,12 @@ https://console.developers.google.com/flows/enableapi?apiid=calendar.
       public handleItemClick(event: SyntheticEvent<any>, name: string): void {
         if (name === 'sign-in') {
           ApiCalendar.handleAuthClick();
```
Comment (Kubessandra) [inline README.md:77]:
```suggestion
          ApiCalendar.handleAuthClick()
```

File: README.md [hunk 4]
```diff
@@ -74,6 +75,12 @@ https://console.developers.google.com/flows/enableapi?apiid=calendar.
...
+          .then(() => {
+            'sign in succesful!'
```
Comment (Kubessandra) [inline README.md:79]:
```suggestion
            console.log('sign in succesful!');
```

File: README.md [hunk 5]
```diff
@@ -74,6 +75,12 @@ https://console.developers.google.com/flows/enableapi?apiid=calendar.
...
+          })
+          .catch((e) => {
+            `sign in failed ${e}`
```
Comment (Kubessandra) [inline README.md:82]:
```suggestion
            console.error(`sign in failed ${e}`);
```

**Review by Kubessandra — CHANGES_REQUESTED** [review]

**Review by Kubessandra — APPROVED** [review]

**Pull request merged**